README.md

```diff
@@ -29,3 +29,126 @@ OpenEBS/project
 ```
 
 > Note: Developer should keep `integration_test` completely decoupled from the rest of the project packages.
+
+## Instantiation
+
+Developer has to instantiate CITF using `NewCITF(configFilePath)` which will initialize the CITF instance with all the required configurations. 
+
+CITF has four components:- 
+- Environment - To Setup or TearDown the platform such as minikube, GKE, AWS etc.
+- K8S - K8S will have Kubernetes ClientSet & Config.
+- Docker - Docker will be used for docker related operations.
+- DebugEnabled - for verbose log.
+
+> Currently CITF supports minikube only.
+
+Developer can pass environment according to their requirements.
+
+By default it will take Minikube as environment.
+
+## Confiuration
+
+To configure the environment of CITF, there are three ways:-
+ - [Environment Variable](#environment-variable)
+ - [Config File](#config-file)
+ - [Default Config](#default-config)
+
+### Environment Variable
+
+At the time of instantiation of CITF, developer can set an environment variable. Environment variable name should be `CITF_CONF_ENVIRONMENT`.
+
+For example:- `export CITF_CONF_ENVIRONMENT = minikube`
+
+### Config File
+If environment variable is not set then developer can pass environment using config file. The file should be in `yaml` format. 
+
+For example:- config.yaml
+
+```
+Environment: minikube
+```
+
+### Default Config
+
+If environment variable and config file are not present, then CITF will take default environment which is minikube.
+
+<details>
+<summary><b>Setting up the Platform</b></summary>
```

Comment:
In this section you are talking about everything that we offer for environment (`Setup`, `Status`, `Teardown`) not just `setting up`. See if it can be improved.

Reply:
Here, we can use platform specifics !!
@a4abhishek Is it good ?